### PR TITLE
refactor(structures): rename create* functions

### LINKED ISSUE #453: Rename: `createGuildRole` to `createRole`
Since only guilds have permissions therefore have roles the `Guild` in the function name is redundant and should be removed.

### PR DESCRIPTION
**Pull request by itohatweb**
**Please describe the changes this PR makes and why it should be merged:**
- Renamed `create*` structure functions to `create*Struct`.
- Renamed the `createServer` function to `createGuild`.

Closes #453

**Status**

- [ ] Code changes have been tested against the Discord API, or there are no code changes

**Semantic versioning classification:**

- [x] This PR changes the library's interface (methods or parameters added)
  - [x] This PR includes breaking changes (methods removed or renamed, parameters moved or removed)
- [ ] This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

**Review by Skillz4Killz — APPROVED** [review]

**Review by ayntee — DISMISSED** [review]

**Comment by ayntee** [comment]
Can you also rename `createGuildRole` to `createRole`? 

Add this at the end of the body of the issue:
```
Closes #453
```

File: src/api/controllers/interactions.ts [hunk 1]
```diff
@@ -11,16 +11,21 @@ export async function handleInternalInteractionCreate(data: DiscordPayload) {
...
+  let member = await cacheHandlers.get("members", payload.member.user.id);
+  if (!member) {
+    const memberStruct = await structures.createMemberStruct(
+      payload.member,
+      payload.guild_id,
+    );
+    await cacheHandlers.set("members", memberStruct.id, memberStruct);
+
+    member = memberStruct;
+  }
```
Comment (ayntee) [inline src/api/controllers/interactions.ts:23]: Can you revert this back?

Comment (itohatweb) [reply src/api/controllers/interactions.ts:23]: yes one sec

**Review by ayntee — APPROVED** [review]

**Review by ayntee — APPROVED** [review]
LGTM

**Review by Skillz4Killz — APPROVED** [review]

**Pull request merged**